### DISCUSSION TITLE
Do applicationDidLaunch and applicationDidTerminate trigger multiple times for the same application running multiple times in parallel?

### DISCUSSION THREAD

**Discussion by nekename** [elgatosf/elgato#60] [General] [answered]
Quick question about the applicationDidLaunch and applicationDidTerminate: say if I'm running an exe/app bundle twice, does the event get sent twice or only once? (assuming the first one is still running when I launch the second one)

Or, for example, Chrome, where Chrome (Renderer), Chrome (Helper), etc all come under chrome.exe, how many events does that send when Chrome launches? Would it send like 5 events when they all terminate, or only one?

**Comment by GeekyEggo** [comment, marked as answer]
Hi @ninjadev64, having tested this locally, I can confirm that the `applicationDidLaunch` and `applicationDidTerminate` events occur every time a monitored application launches or terminates. For example.
1. With the following manifest
```json
"ApplicationsToMonitor": {
    "windows": [ "Code.exe" ]
}
```
1. When I launch VS Code, the `applicationDidLaunch` event is emitted.
1. Launching a new window of VS Code emits a new `applicationDidLaunch` event.

This is true for `applicationDidTerminate` as well.

**Reply by nekename** [reply]
Oh, thanks! If you have a Mac, would you mind testing the same, because Chromium-based applications start multiple processes for one window on macOS, such as Chrome (Helper), Chrome (Renderer), etc... If not, no worries!
